backends/core/workers.py

```diff
@@ -760,3 +764,76 @@ def _execute(self):
     self.retry(request.execute)()
     job_name = '%s/jobs/%s' % (project_id, self._ml_job_id)
     self._enqueue('MLWaiter', {'job_name': job_name}, 60)
+
+
+class MeasurementProtocolException(WorkerException):
+  """Measurement Protocol execution exception."""
+  pass
+
+
+class MeasurementProtocolWorker(Worker):
+  """Abstract Measurement Protocol worker."""
+
+  def _send_hit(self, user_agent='CRMint / 0.1', **kwargs):
+    """Send a measurement protocol hit.
+
+    Arguments:
+      user_agent string representing the client User Agent.
+      kwargs dictionary of key/values to pass to the Measurement
+          Protocol endpoint.
+
+    Raises: MeasurementProtocolException if the HTTP request fails.
+    """
+    headers = {'user-agent': user_agent}
+    payload = {'v': 1}
+    payload.update(kwargs)
+
+    req = requests.post('https://www.google-analytics.com/collect',
+                        headers=headers,
+                        data=payload)
+
+    if req.status_code != requests.codes.ok:
+      raise MeasurementProtocolException('Failed to send event hit with status'
+                                         'code (%s) and parameters: %s'
+                                         % (req.status_code, kwargs))
+
+
+class BQToMeasurementProtocol(BQWorker, MeasurementProtocolWorker):
+  """Worker to import data through Measurement Protocol"""
+
+  PARAMS = [
+      ('bq_project_id', 'string', False, '', 'BQ Project ID'),
+      ('bq_dataset_id', 'string', True, '', 'BQ Dataset ID'),
+      ('bq_table_id', 'string', True, '', 'BQ Table ID'),
+  ]
+
+  def _process_query_results(self, query_data):
+    """Sends event hits for this chunk of data."""
+    fields = [f.name for f in query_data.query_result.schema]
+    for row in query_data:
+      data = dict(zip(fields, row))
+      try:
+        self.retry(self._send_hit)(**data)
+      except MeasurementProtocolException as inst:
+        self.log_error(inst.message)
+
+  def _execute(self):
+    # Retrieves data from BigQuery.
+    client = self._get_client()
+    query = 'SELECT * FROM `%s.%s.%s`' % (
```

Comment:
Please, use `self._table.fetch_data()` as in method `_infer_audiences()` of `GAAudiencesUpdater` class instead of issuing a `SELECT * FROM …` query. It will save BigQuery from creating intermediate tables for query results. Also, please, use `_bq_setup()` method in previous line to get BigQuery client and BigQuery table.

Reply:
Done. I also rewrote the tests to match the new way of requesting this.